src/ring_theory/noetherian.lean

```diff
@@ -24,6 +24,43 @@ theorem fg_def {s : submodule α β} :
   exact ⟨t, rfl⟩
 end⟩
 
+/-- Nakayama's Lemma. Atiyah-Macdonald 2.5, Eisenbud 4.7, Matsumura 2.2, Stacks 10.19.1 (00DV) -/
+theorem exists_sub_one_mem_and_smul_eq_zero_of_fg_of_le_smul {R : Type*} [comm_ring R]
+  {M : Type*} [add_comm_group M] [module R M]
+  (I : ideal R) (N : submodule R M) (hn : N.fg) (hin : N ≤ I • N) :
```

Comment:
Can you also give the corollary for finitely generated modules. (I.e., the version that does not mention submodules.) I guess that is a statement that will be applied quite often.

Reply:
How would you state it?